rudder-core/src/main/scala/com/normation/cfclerk/domain/Constraint.scala

```diff
@@ -192,12 +187,12 @@ final case object LinuxDerivedPasswordVType extends DerivedPasswordVType { overr
 
 case object BooleanVType extends VTypeConstraint {
   override val name = "boolean"
-  override def getTypedValue(value:String, forField:String) : Box[Any] = {
-      try {
+  override def getFormatedValidated(value:String, forField:String, escapeString: String => String) : Box[String] = {
```

Comment:
This is the cause of the issue in templates, the value was previous a boolean, not a string, then when used in conditional in string template, if considers truthy anything that has a value or check the boolean value if it's a boolean:

> IF actions test the presence or absence of an attribute unless the object is a Boolean/bool, in which case it tests the attribute for true/false. The only operator allowed is "not" and means either "not present" or "not true". For example, "$if(!member)$...$endif$".